extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeCommentsExtractor.java

```diff
@@ -95,36 +102,46 @@ public InfoItemsPage<CommentsInfoItem> getPage(final Page page)
             throw new IllegalArgumentException("Page doesn't contain an URL");
         }
 
-        final Response response = getDownloader().get(page.getUrl());
-
         JsonObject json = null;
-        if (response != null && !Utils.isBlank(response.responseBody())) {
+        final CommentsInfoItemsCollector collector = new CommentsInfoItemsCollector(getServiceId());
+        final long total;
+        if (page.getBody() == null) {
+            final Response response = getDownloader().get(page.getUrl());
+            if (response != null && !Utils.isBlank(response.responseBody())) {
+                try {
+                    json = JsonParser.object().from(response.responseBody());
+                } catch (final Exception e) {
+                    throw new ParsingException("Could not parse json data for comments info", e);
+                }
+            }
+            if (json != null) {
+                PeertubeParsingHelper.validate(json);
+                if (isReply() || json.has(CHILDREN)) {
+                    total = json.getArray(CHILDREN).size();
+                    collectRepliesFrom(collector, json);
+                } else {
+                    total = json.getLong(TOTAL);
+                    collectCommentsFrom(collector, json);
+                }
+            } else {
+                throw new ExtractionException("Unable to get PeerTube kiosk info");
+            }
+        } else {
```

Comment:
Aren't these `else` statements unneeded?

Reply:
Do you want me to add another return above? I could do that, but IMO if/else is better here.